<a href="https://colab.research.google.com/github/sirakik/japan_graph/blob/master/SEKAI_DATA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

「地球には国境などどこにもない」   ので，細かいことは考えない.

###あまりよくないソース

国境（個人のブログ）：　https://misunderstanding.hatenadiary.org/entry/20090305/1236185179  
ISO　3166-1コード一覧 https://qiita.com/tao_s/items/32b90a2751bfbdd585ea  

In [106]:
import csv
!wget -q https://raw.githubusercontent.com/sirakik/japan_graph/master/the_creation/border.csv -O border.csv
with open('border.csv') as f:
    reader = csv.reader(f)
    border = [row for row in reader]

!wget -q https://raw.githubusercontent.com/sirakik/japan_graph/master/the_creation/iso_rename.csv -O iso.csv
with open('iso.csv') as f:
    reader = csv.reader(f)
    iso = [row for row in reader]

!wget -q https://raw.githubusercontent.com/sirakik/japan_graph/master/the_creation/captal.csv -O capital.csv
with open('capital.csv') as f:
    reader = csv.reader(f)
    capital = [row for row in reader]

In [107]:
print(len(border))#194カ国(おそらく国連加盟国＋バチカン)　国境データを優先
print(len(iso))    #250カ国
print(len(capital))

194
251
200


国境データ：194カ国(おそらく国連加盟国＋バチカン)　  
ISOデータ：251カ国  

多くの国に認められていないような国家の情報は，ほとんど取得できないと想像.  
ノード数は,国境データを利用



国のindex(0~193)はISOデータを利用（50音順）


In [86]:
print(border[0])
print(iso[1])

['ロシア', 'アゼルバイジャン', 'ウクライナ', 'エストニア', 'カザフスタン', 'グルジア', '中華人民共和国', '朝鮮民主主義人民共和国', 'ノルウェー', 'フィンランド', 'ベラルーシ', 'ポーランド', 'モンゴル', 'ラトビア', 'リトアニア']
['アイスランド', 'Iceland', '352', 'ISL', 'IS', '北ヨーロッパ', 'ISO 3166-2:IS']


データによって国名が違うのため，手作業で変更(ISOデータの日本語名を変換)

ISOにコソボが無いが, 国境データにはある.　（コソボは国連未加入）
ISOデータにコソボデータを追加


欲しいデータ
['国名(英語)', '英3字', '首都の緯度経度(後で)', '[隣接国]']

In [108]:
country_list = []


for row in iso:
    country_jp = row[0]
    
    for row_border in border:
        if country_jp == row_border[0]:
            neighbor = [i for i in row_border[1:] if i != '']

            for i, nie_country in enumerate(neighbor):
                for row_ in iso:
                    if nie_country == row_[0]:
                        neighbor[i] = row_[1]

            country_list.append([row[1], row[3], row[4], neighbor])
            break

In [95]:
print(country_list)

[['Iceland', 'ISL', 'IS', []], ['Ireland', 'IRL', 'IE', ['United Kingdom']], ['Azerbaijan', 'AZE', 'AZ', ['Iran', 'Georgia', 'Armenia', 'Russian Federation']], ['Afghanistan', 'AFG', 'AF', ['Iran', 'Uzbekistan', 'Tajikistan', 'China', 'Turkmenistan', 'Pakistan']], ['United States', 'USA', 'US', ['Canada', 'Mexico']], ['United Arab Emirates', 'ARE', 'AE', ['Oman', 'Qatar', 'Saudi Arabia']], ['Algeria', 'DZA', 'DZ', ['Tunisia', 'Niger', 'Mali', 'Mauritania', 'Morocco', 'Libya']], ['Argentina', 'ARG', 'AR', ['Uruguay', 'Chile', 'Paraguay', 'Bolivia', 'Brazil']], ['Albania', 'ALB', 'AL', ['Greece', 'Kosovo', 'Macedonia', 'Montenegro']], ['Armenia', 'ARM', 'AM', ['Iran', 'Azerbaijan', 'Georgia', 'Turkey']], ['Angola', 'AGO', 'AO', ['Congo', 'Congo', 'Zambia', 'Namibia']], ['Antigua and Barbuda', 'ATG', 'AG', []], ['Andorra', 'AND', 'AD', ['Spain', 'France']], ['Yemen', 'YEM', 'YE', ['Oman', 'Saudi Arabia']], ['United Kingdom', 'GBR', 'GB', ['Ireland', 'Cyprus', 'Spain']], ['Israel', 'ISR', 

首都の緯度経度を格納

In [109]:
for i, cou in enumerate(country_list):
    code = cou[2]
    for cap in capital:
        if code == cap[0]:
            country_list[i][2] = [cap[1], cap[2]]
            break

In [110]:
print(country_list)

[['Iceland', 'ISL', ['64.1265206', '-21.8174392'], []], ['Ireland', 'IRL', ['53.3498053', '-6.2603097'], ['United Kingdom']], ['Azerbaijan', 'AZE', ['40.4092617', '49.8670924'], ['Iran', 'Georgia', 'Armenia', 'Russian Federation']], ['Afghanistan', 'AFG', ['34.5553494', '69.207486'], ['Iran', 'Uzbekistan', 'Tajikistan', 'China', 'Turkmenistan', 'Pakistan']], ['United States', 'USA', ['38.9071923', '-77.0368707'], ['Canada', 'Mexico']], ['United Arab Emirates', 'ARE', ['24.453884', '54.3773438'], ['Oman', 'Qatar', 'Saudi Arabia']], ['Algeria', 'DZA', ['36.753768', '3.0587561'], ['Tunisia', 'Niger', 'Mali', 'Mauritania', 'Morocco', 'Libya']], ['Argentina', 'ARG', ['-34.6036844', '-58.3815591'], ['Uruguay', 'Chile', 'Paraguay', 'Bolivia', 'Brazil']], ['Albania', 'ALB', ['41.3275459', '19.8186982'], ['Greece', 'Kosovo', 'Macedonia', 'Montenegro']], ['Armenia', 'ARM', ['40.1791857', '44.4991029'], ['Iran', 'Azerbaijan', 'Georgia', 'Turkey']], ['Angola', 'AGO', ['-8.8399876', '13.2894368'], 

In [111]:
from google.colab import drive
drive.mount('./gdrive')

Mounted at ./gdrive


In [122]:
!ls gdrive/My\ Drive
!ls

'Colab Notebooks'   Japan   MPRG_lecture
border.csv  capital.csv  gdrive  iso.csv  sample_data


In [127]:
drive_root_dir="./gdrive/My Drive/"

with open(drive_root_dir + 'world.csv', 'w') as f:
    writer = csv.writer(f)
    writer.writerows(country_list)